libclambcc/ClamBCLogicalCompiler/ClamBCLogicalCompiler.cpp

```diff
@@ -1644,14 +1648,48 @@ bool ClamBCLogicalCompiler::compileVirusNames(Module &M, unsigned kind)
             continue;
         }
         assert(CS.arg_size() == 2 && "setvirusname has 2 args");
+#else
+        CallInst * pCallInst = llvm::cast<CallInst>(I);
+        if (nullptr == pCallInst){
+            assert (0 && "NOT sure how this is possible");
+            continue;
+        }
+
+        if (F != pCallInst->getCalledFunction()){
+
+            llvm::errs() << "<" << __FUNCTION__ << "::" << __LINE__ << ">NOT SURE HOW THIS IS POSSIBLE<END>\n";
+
+            /*Not sure how this is possible, either*/
+            printDiagnostic("setvirusname can only be directly called",
+                            pCallInst);
+            Valid = false;
+            continue;
+        }
+#endif
+
+        if (2 != pCallInst->arg_size()){
+            printDiagnostic("setvirusname has 2 args", pCallInst);
+            Valid = false;
+            continue;
+        }
+
         std::string param;
         llvm::StringRef sr;
+#if 0
         Value *V    = CS.getArgument(0);
+#else
+#endif
+        Value * V = llvm::cast<Value>(pCallInst->arg_begin());
```

Comment:
I imagine you actually wanted this inside the else.  But since the plan is to delete all the dead code in the big-PR,... whatever. :)